pkg/commands/crypto.go

```diff
@@ -0,0 +1,150 @@
+package commands
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"io"
+	"math/big"
+	mathRand "math/rand"
+)
+
+// encrypt text using the key
+func encrypt(text, key string) ([]byte, error) {
+	gcm, err := createGCM(key)
+	if err != nil {
+		return nil, err
+	}
+
+	nonce := make([]byte, gcm.NonceSize())
+	if _, err = io.ReadFull(rand.Reader, nonce); err != nil {
+		return nil, err
+	}
+
+	return gcm.Seal(nonce, nonce, []byte(text), nil), nil
+}
+
+// decrypt a string output from key
+func decrypt(data []byte, key string) (string, error) {
+	gcm, err := createGCM(key)
+	if err != nil {
+		return "", err
+	}
+
+	nonceSize := gcm.NonceSize()
+	if len(data) < nonceSize {
+		return "", errors.New("ciphertext too short")
+	}
+
+	nonce, ciphertext := data[:nonceSize], data[nonceSize:]
+	out, err := gcm.Open(nil, nonce, ciphertext, nil)
+	if err != nil {
+		return "", err
+	}
+
+	return string(out), nil
+}
+
+func createGCM(key string) (cipher.AEAD, error) {
+	sha256Key := sha256.Sum256([]byte(key))
+	aesCipher, err := aes.NewCipher(sha256Key[:])
+	if err != nil {
+		return nil, err
+	}
+
+	return cipher.NewGCM(aesCipher)
+}
+
+// encryptWithKeyPair encrypts a string using a public key
+func encryptWithKeyPair(publicKey, data string) ([]byte, error) {
```

Comment:
this method name is not quite right: I believe a clearer name would be 'encryptWithPublicKey'.